blog/content/post/2025-05-07-dasharo-ci.md

```diff
@@ -0,0 +1,293 @@
+  - uefi
+categories:
+  - Firmware
+  - Security
+
+---
+
+## Introduction
+
+From the moment you press the power button, the security of your device begins
+with your boot firmware. It plays a critical role in establishing the
+trustworthiness of your device.
+
+Dasharo is our open-source firmware distribution based on coreboot, which is
+a lightweight initial bootloader that initializes the hardware platform before
+passing control to a secondary bootloader. In Dasharo, the secondary bootloader
+is typically TianoCore EDK2, which is an implementation of UEFI.
+
+One of the responsibilities of coreboot is to provide microcode updates for the
+processor. Microcode updates are critical for firmware security, as they contain
+fixes and mitigations for many classes of exploits, like
+[Spectre and Meltdown](https://meltdownattack.com/).
+
+On the UEFI side, there is UEFI Secure Boot, which is a method for
+cryptographically verifying OS bootloaders before executing them. One of the
+components of UEFI Secure Boot is a revocation database, which is called DBX.
+It contains hashes of revoked binaries, such as bootloaders with known security
+vulnerabilities, like [GRUB and the BootHole exploit](https://eclypsium.com/blog/theres-a-hole-in-the-boot/).
+This mechanism provides a way to revoke binaries that were previously considered
+trusted.
+
+When it comes to overall firmware security trust chain, microcode sits at the
+very beginning, and UEFI Secure Boot sits at the very end, just before the OS
+starts to load. This is why updating these components regularly is crucial for
+platform security, and why we need automatic checks that ensure these components
+are up to date.
+
+In this blog post we'll explore how simple GitHub actions can ensure that these
+critical security components are always up-to-date, helping us deliver a secure
+firmware solution for our users.
+
+## The components
+
+Both Intel Microcode binaries and the UEFI DBX file are publicly available, by
+Intel and the UEFI forum respectively.
+
+Intel Microcode updates are available at Intel's GitHub page:
+[link](https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files).
+
+From their readme:
+
+> The Intel Processor Microcode Update (MCU) Package provides a mechanism to
+> release updates for security advisories and functional issues, including
+> errata.
+
+The microcode is typically provided by coreboot in a Firmware Interface Table
+(FIT). This table is parsed before the x86 cores begin to execute any
+instructions, which means that the update is performed before coreboot has even
+had a chance to start.
+
+In addition, modern CPUs often depend on ucode updates to _function at all_.
+Due to the amount of erratas, a processor may simply refuse to do anything if
+a microcode update is not provided.
+
+The OS is also able to load microcode updates, but due to how late in the boot
+process that is done, it might be too late to patch some security
+vulnerabilities or erratas.
```

Comment:
Aren't there some advisories saying certain mitigation can be applied while loading at early boot chain stage?

Reply:
Yes, added https://github.com/3mdeb/news-and-ideas/pull/659/commits/382a588a9c03fc0e27e6e098d34e6975789c4455

Reply:
This fix is good, but give one example based on [this comment](https://github.com/3mdeb/news-and-ideas/pull/659#discussion_r2081097630) I would expect 2-3 per DBX and 2-3 per microcode described at least in 1-2 sentences with link.

Reply:
Added more examples in https://github.com/3mdeb/news-and-ideas/pull/659/commits/f0fe62c23ec02b1edbc458d3ed1ebc12ccce2b53

Reply:
Yes, but you didn't applied:

> described at least in 1-2 sentences with link

More examples is good, but do I have to really go behind those links to justify value of what you explaining? I would like to read more when it would be interesting enough. So please give me 1-2 sentences so I can decide if your story sticks.

Reply:
Added a bit more https://github.com/3mdeb/news-and-ideas/pull/659/commits/7a7acbb6d7d495e9a26e39113cfe06891370af57

Reply:
@mkopec love it, thank you.